### PR TITLE
[CP-2425] Refactor: Organized usage of history and improved routing mock

### PR DESCRIPTION
**Pull request by dkarski**
JIRA Reference: [CP-2425]

### :memo: Description ️

- Simplification of the react-router-dom mock by adding a mocking file.
- Migration of the Router provider from a single test to a wrapper helper.
- Reduction of history complexity by utilizing a history instance.
- Refactoring of the application structure in core/core.

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2425]: https://appnroll.atlassian.net/browse/CP-2425?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/core/components/apps-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { Mode } from "Core/__deprecated__/common/enums/mode.enum"
+import ErrorApp from "Core/core/components/apps/error-app.component"
+import HelpApp from "Core/core/components/apps/help-app.component"
+import LicenseApp from "Core/core/components/apps/license-app.component"
+import TermsOfServiceApp from "Core/core/components/apps/terms-of-service-app.component"
+import PrivacyPolicyApp from "Core/core/components/apps/privacy-policy-app.component"
+import SarApp from "Core/core/components/apps/sar-app.component"
+import BaseAppContainer from "Core/core/components/apps/base-app/base-app-container.component"
+
+const AppsSwitch = () => {
```
Comment (lkowalczyk87) [inline libs/core/core/components/apps-switch.tsx:16]: looks like something that should be resolved via routing

Comment (lkowalczyk87) [inline libs/core/core/components/apps-switch.tsx:16]: ensure that component is not re-rendered every 10 sec.

File: libs/core/core/components/apps/help-app.component.tsx [hunk 2]
```diff
@@ -4,34 +4,26 @@
...
 
 const devModeEnabled = flags.get(Feature.DeveloperModeEnabled)
 
-const HelpApp: FunctionComponent<Props> = ({
-  history,
-  saveToStore,
-  getStoreData,
-}) => {
+const HelpApp: FunctionComponent = () => {
+  const saveToStore = async (normalizeData: QuestionAndAnswer) =>
```
[the remark this answers is not in the document]
Comment (dkarski) [reply libs/core/core/components/apps/help-app.component.tsx]: https://github.com/mudita/mudita-center/pull/1674/commits/3825491cce131631b44b0cb3d0b7c6defb83d9c3

File: libs/core/core/components/apps-switch.tsx (hunk 1 above)
Comment (dkarski) [reply libs/core/core/components/apps-switch.tsx:16]: https://github.com/mudita/mudita-center/assets/17147149/5a3fa921-6fdf-42a4-824b-8cec775436e4

Comment (dkarski) [reply libs/core/core/components/apps-switch.tsx:16]: Your idea about using React Routing in AppsSwitch is getting its own task (https://appnroll.atlassian.net/browse/CP-2461). Thanks for the suggestion!

**Pull request merged**